pkg/registry/scheduling/rest/storage_scheduling.go

```diff
@@ -50,42 +50,62 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(scheduling.GroupName, legacyscheme.Scheme, legacyscheme.ParameterCodec, legacyscheme.Codecs)
 
 	if apiResourceConfigSource.VersionEnabled(schedulingapiv1alpha1.SchemeGroupVersion) {
-		apiGroupInfo.VersionedResourcesStorageMap[schedulingapiv1alpha1.SchemeGroupVersion.Version] = p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
+		if storage, err := p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter); err != nil {
+			return genericapiserver.APIGroupInfo{}, false
```

Comment:
this is incorrect... an error case silently disabling the group/version but letting the server start is not what we want

Reply:
applies to all instances of this pattern in this file

Reply:
InstallAPIs doesn't return error at the moment.

I am changing the return value to error

Reply:
this returns false but does not return the error. if we want to make this a fatal error, the error information must be returned

Reply:
Please note the following block in master.go:
```
	restStorageProviders := []RESTStorageProvider{
```
if error is to be returned from NewRESTStorage, looks like a dozen classes need to be changed.
Please confirm that this is the way to go.

Reply:
yes, if those can error, and we want to propagate the error, all signatures need changing

Reply:
I plan to add third return value for the error and keep the second return value as boolean.